libs/traits/src/liquidity_pools.rs

```diff
@@ -31,11 +33,20 @@ pub trait LPEncoding: Sized {
 	/// Creates an empty message.
 	/// It's the identity message for composing messages with pack_with
 	fn empty() -> Self;
+
+	/// Retrieves the message proof, if any.
+	fn get_message_proof(&self) -> Option<Proof>;
+
+	/// Converts the message into a message proof type.
+	fn to_message_proof(&self) -> Self;
```

Comment:
I never was a big fan of the names that I picked for these haha. Your suggestion is definitely better, thank you!

Reply:
Maybe `message_hash()` is more exact than `proof_hash()` because, actually, it's just a hash that does nothing to do with proof. Later, the proof uses that hash, but that's all. WDYT?

Reply:
`message_proof` would be the best fitting for this case, `message_hash` can also mean other things IMO.

How about - `message_proof` and `to_proof_message`?

Reply:
Mmm... But the proof itself is not a hash? More specifically, a `keccak_256` hash? Later, that hash is used as proof by the message proof. Probably this would mean we should rename the type `Proof` as `MessageHash` in order to split clearly both concepts.

Reply:
This is really not that simple picking well descriptive names! I actually liked having the `get_message_proof` because it returns an option which is usually the case for getters.

I will throw another option in the fictional bin of options: Keep the first one slightly reduced `get_proof` and name the other one `to_proof_message` as proposed by @cdamian.

Reply:
Ok, renamed this.

Reply:
Some notes below regarding this